### PR TITLE
Convert to go modules

### PR DESCRIPTION
**Pull request by rodrigc**
The following Portworx projects use go modules:

* https://github.com/portworx/sched-ops/blob/master/go.mod
* https://github.com/portworx/torpedo/blob/master/go.mod
* https://github.com/portworx/spawn/blob/master/go.mod

Kubernetes 1.15 and higher use go modules:

* https://github.com/kubernetes/kubernetes/blob/release-1.15/go.mod

It will be easier to interoperate between this projects if `libopenstorage/operator` also uses go modules.

### CODE REVIEW

**Comment by thiguetta** [comment]
in general, it looks good to me, don't know if it might be a concern but we have other projects in libopenstorage using dep, such as:
https://github.com/libopenstorage/stork
https://github.com/libopenstorage/autopilot-api
https://github.com/libopenstorage/gossip

**Comment by rodrigc** [comment]
Those can be converted over time.  I have an old PR for stork for example that needs to be brought up to date:
https://github.com/libopenstorage/stork/pull/296

**Comment by rodrigc** [comment]
@thiguetta I updated my PR to stork for go modules: https://github.com/libopenstorage/stork/pull/296

File: hack/update-codegen.sh [hunk 1]
```diff
@@ -21,7 +21,22 @@ set -o pipefail
...
+function cleanup {
+  mv ${CODEGEN_PKG}/generate-groups.sh.orig ${CODEGEN_PKG}/generate-groups.sh
+}
+trap cleanup EXIT
+
+go install ./${CODEGEN_PKG}/cmd/{defaulter-gen,client-gen,lister-gen,informer-gen,deepcopy-gen}
+
+
 # generate the code with:
+chmod +x ${CODEGEN_PKG}/generate-groups.sh
```
Comment (piyush-nimbalkar) [inline hack/update-codegen.sh:39]: Or we could just do `bash ${CODEGEN_PKG}/genere....`

Comment (rodrigc) [reply hack/update-codegen.sh:39]: Sure we could do that, that's probably better.

Comment (rodrigc) [reply hack/update-codegen.sh]: Fixed

**Comment by harsh-px** [comment]
No longer required. Operator is already using go modules.

**Pull request closed without merging**